test-runner-frontend/src/Components/IdCell.js

```diff
@@ -29,16 +30,32 @@ export default function IdCell({ value, hasCheckbox, onChange, status }) {
     <>
       {hasCheckbox ? (
         <Stack direction="row" spacing={2}>
-          <Checkbox value={value} onChange={() => onChange(value)} />
+          {status === 'FINISHED' ? (
+            <Checkbox data-testid="enabledCheckbox" value={value} onChange={() => onChange(value)} />
+          ) : (
+            <Checkbox data-testid="disabledCheckbox" disabled={true} value={value} onChange={() => onChange(value)} />
+          )}
           <Stack direction="row" spacing={2} style={{ marginTop: '1.8%' }}>
             <IconComponent sx={{ color: iconColor }} />
-            <div>{value}</div>
+            {status === 'FINISHED' ? (
+              <Link to="/report" state={{ from: row }} style={{ textDecoration: 'none', color: 'black' }}>
+                <div style={{ textDecoration: 'underline' }}>{value}</div>
+              </Link>
+            ) : (
+              <div style={{ textDecoration: 'none' }}>{value}</div>
+            )}
```

Comment:
because we're beginning to build up a lot of duplicate children here and on L52:L58, we should store these contents as a variable under where we decide our colours and icons based off the status.

Reply:
Done.